src/borg/testsuite/helpers.py

```diff
@@ -342,16 +343,36 @@ def test_chunkerparams():
         ChunkerParams("fixed,%d,%d" % (4096, MAX_DATA_SIZE + 1))  # too big header size
 
 
+class RemoveDotdotPrefixesTestCase(BaseTestCase):
+    def test(self):
+        self.assert_equal(remove_dotdot_prefixes("."), ".")
+        self.assert_equal(remove_dotdot_prefixes(".."), ".")
+        self.assert_equal(remove_dotdot_prefixes("/"), ".")
+        self.assert_equal(remove_dotdot_prefixes("//"), ".")
+        self.assert_equal(remove_dotdot_prefixes("foo"), "foo")
+        self.assert_equal(remove_dotdot_prefixes("foo/bar"), "foo/bar")
+        self.assert_equal(remove_dotdot_prefixes("/foo/bar"), "foo/bar")
+        self.assert_equal(remove_dotdot_prefixes("../foo/bar"), "foo/bar")
```

Comment:
Stripping "./" and "/" is the usual thing borg does.

When listing (or extracting), we'll **know** it either refers to "/" or to "." (which is already ambiguous, but guess it is what tar does and attic/borg copied that behaviour).

If we just strip arbitrary amounts of "../", we basically have no clue any more about the directory level on which it was.

Reply:
> When listing (or extracting), we'll know it either refers to "/" or to "." (which is already ambiguous, but guess it is what tar does and attic/borg copied that behaviour).

Talking of tar, GNU tar prints a warning when it strips anything from the path. The same could be considered for borg except that we shouldn't complain when a single / at the beginning of a path is removed.

```
user@dev-borg:~/src/borg$ tar cf tmp.tar ../borg-upstream/
tar: Removing leading `../' from member names
```

Reply:
Well, if you assume that the cwd at the time of the borg invocation is known (because it is part of the backup script), we know what `.` referred to.

But if borg gets fed all sorts of `../`*N and it removes them all, then we can't reconstruct the location any more. In case of db dumps it might not matter, but in many other cases it might.

BTW, macOS tar:
```
tw@mba2020 ~ % tar cvf foo.tar ../tw/foo.txt ../../etc/hosts
a ../tw/foo.txt
a ../../etc/hosts
tw@mba2020 ~ % tar tvf foo.tar 
-rw-r--r--  0 tw     staff     361 Jul 21  2021 ../tw/foo.txt
-rw-r--r--  0 root   wheel     264 Dec 26 18:59 ../../etc/hosts
```